### PR TITLE
Second Module item complete

### PR DESCRIPTION
**Pull request by SartiMau**

### CODE REVIEW

File: data/src/main/java/com/globant/equattrocchio/data/ImagesServicesImpl.java [hunk 1]
```diff
@@ -18,57 +22,71 @@
...
+            public void onFailure(Call<Result> call, Throwable t) {
+                //todo: update the UI with a connection error message
                 observer.onError(t);
             }
         });
+
+
+    }
+
+    private List<ImageEntity> parseImageDataDomain(Result result) {
```
Comment (fanky10) [inline data/src/main/java/com/globant/equattrocchio/data/ImagesServicesImpl.java]: just rename it to:  transform

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java [hunk 2]
```diff
@@ -5,83 +5,82 @@
...
 
 public class ImagesPresenter {
 
     private ImagesView view;
-//    private GetLatestImagesUseCase getLatestImagesUseCase;
-    private GetJsonUseCase getJsonUseCase;
 
-//    public ImagesPresenter(ImagesView view, GetLatestImagesUseCase getLatestImagesUseCase) {
+//    This is from the second Module's item
+//    private GetJsonUseCase getJsonUseCase;
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java]: remove it, it's fine not having code for second module, there's already a PR for that

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java [hunk 3]
```diff
@@ -5,83 +5,82 @@
...
+
+//    This is from the third Module's item
+    private GetLatestImagesUseCase getLatestImagesUseCase;
+
+//    This is from the second Module's item
+//    public ImagesPresenter(ImagesView view, GetJsonUseCase getJsonUseCase) {
 //        this.view = view;
-//        this.getLatestImagesUseCase = getLatestImagesUseCase;
+//        this.getJsonUseCase = getJsonUseCase;
 //    }
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java]: this too, please remove

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java [hunk 4]
```diff
@@ -5,83 +5,82 @@
...
             @Override
             public void onComplete() {
 
             }
-        }, null);
-
-
+        },null);
 
         //todo acá tengo que llamar a la domain layer para que llame a la data layer y haga el llamdo al servicio
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java]: remove comments in Spanish

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    public RecyclerViewAdapter(Context context, List<ImageEntity> images) {
+        this.context = context;
+        this.images = images;
+    }
+
+    @NonNull
+    @Override
+    public RecyclerViewViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        //crea el ViewHolder
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: remove this

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.card,parent,false);
+        return new RecyclerViewViewHolder(itemView);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerViewViewHolder holder, int position) {
+        //carga los datos, popula la lista
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: remove this as well

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        holder.getCardImageId().setText("Image ID: " + images.get(position).getId());
+        Glide
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: format correctly

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/base/MainActivity.java [hunk 8]
```diff
@@ -23,11 +23,13 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-//        GetLatestImagesUseCase getLatestImagesUseCase = new GetLatestImagesUseCase(new ImagesServicesImpl());
-//        presenter = new ImagesPresenter(new ImagesView(this),getLatestImagesUseCase);
-        GetJsonUseCase getJsonUseCase = new GetJsonUseCase(new ImagesServicesImpl());
-        presenter = new ImagesPresenter(new ImagesView(this),getJsonUseCase);
+//        This is from the second Module's item
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/base/MainActivity.java]: remove unnecessary second module code

**Review by fanky10 — CHANGES_REQUESTED** [review]
needs rework

File: data/src/main/java/com/globant/equattrocchio/data/ImagesServicesImpl.java (hunk 1 above)
Comment (SartiMau) [reply data/src/main/java/com/globant/equattrocchio/data/ImagesServicesImpl.java]: ok done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java (hunk 4 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java]: ok done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java (hunk 5 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: ok done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java (hunk 6 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: ok done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java (hunk 2 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java]: ok done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java (hunk 3 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java]: ok done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/base/MainActivity.java (hunk 8 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/base/MainActivity.java]: ok done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java (hunk 7 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: ok done

File: presentation/src/main/res/layout/activity_main.xml [hunk 9]
```diff
@@ -14,5 +16,20 @@
...
-            android:text="@string/incoming_json" />
+            android:text="@string/incoming_json"
+            android:enabled="false"/>
+
+        <android.support.v7.widget.RecyclerView
+            android:id="@+id/recycler_view"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            card_view:cardUseCompatPadding="true"
+            card_view:cardElevation="5dp"
```
Comment (gaston-mira-glb) [inline presentation/src/main/res/layout/activity_main.xml]: use dimens please

File: presentation/src/main/res/layout/card.xml [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    xmlns:card_view="http://schemas.android.com/tools">
+
+    <android.support.v7.widget.CardView
+        android:id="@+id/cardView"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_gravity="center"
+        android:elevation="3dp"
```
Comment (gaston-mira-glb) [inline presentation/src/main/res/layout/card.xml]: same here

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/ImagesView.java [hunk 11]
```diff
@@ -30,6 +38,12 @@ public void callServiceBtnPressed() {
     }
 
     public void showError() {
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/ImagesView.java:40]: is error hidden at some point?

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+                .load(images.get(position).getUrl())
+                .into(holder.getCardImage());
+    }
+
+    @Override
+    public int getItemCount() {
+        return images.size();
+    }
+
+    public static class RecyclerViewViewHolder extends RecyclerView.ViewHolder{
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: why public static? it's used only in here, you could just name it ImageViewHolder and make it private to be used as inner class

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+
+        private final TextView cardImageId;
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: just final is fine, you can use it as "public" inside RecyclerViewAdapter

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+import com.globant.equattrocchio.cleanarchitecture.R;
+import com.globant.equattrocchio.domain.enities.ImageEntity;
+
+import java.util.List;
+
+public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.RecyclerViewViewHolder> {
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: I'd rather have a ImagesAdapter than a generic recyclerViewAdapter, please change

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+        private final TextView cardImageId;
+        private final ImageView cardImage;
+
+        public RecyclerViewViewHolder(View parent) {
+            super(parent);
+            cardImageId = itemView.findViewById(R.id.cardImageId);
+            cardImage = itemView.findViewById(R.id.cardImage);
+        }
+
+        public TextView getCardImageId(){
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: remove getters

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/base/MainActivity.java [hunk 16]
```diff
@@ -23,11 +18,8 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-//        GetLatestImagesUseCase getLatestImagesUseCase = new GetLatestImagesUseCase(new ImagesServicesImpl());
-//        presenter = new ImagesPresenter(new ImagesView(this),getLatestImagesUseCase);
-        GetJsonUseCase getJsonUseCase = new GetJsonUseCase(new ImagesServicesImpl());
-        presenter = new ImagesPresenter(new ImagesView(this),getJsonUseCase);
-
+        GetLatestImagesUseCase getLatestImagesUseCase = new GetLatestImagesUseCase(new ImagesServicesImpl());
+        presenter = new ImagesPresenter(new ImagesView(this),getLatestImagesUseCase);
```
Comment (fanky10) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/base/MainActivity.java]: format correctly: presenter = new ImagesPresenter(new ImagesView(this), getLatestImagesUseCase);

File: presentation/src/main/res/layout/card.xml (hunk 10 above)
Comment (fanky10) [inline presentation/src/main/res/layout/card.xml]: avoid hardcoded dp int values, move them to dimens.xml

**Review by fanky10 — CHANGES_REQUESTED** [review]
Please check the naming conventions and code style comments

**Review by gaston-mira-glb — CHANGES_REQUESTED** [review]

**Comment by gaston-mira-glb** [comment]
minor comments

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/ImagesView.java [hunk 17]
```diff
@@ -30,6 +38,12 @@ public void callServiceBtnPressed() {
...
+        tvlabel.setEnabled(true);
         tvlabel.setText(R.string.connection_error);
     }
+
+    public void showImagesInCardView(List<ImageEntity> images) {
+        recyclerView.setAdapter(new RecyclerViewAdapter(this.getContext(), images));
```
Comment (nacho-globant) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/ImagesView.java]: check if getContext is null, and separate in a local variable

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/ImagesView.java (hunk 11 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/ImagesView.java:40]: it was not hidden, now it is hidden when the button is pressed.

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/ImagesView.java (hunk 17 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/ImagesView.java]: ok done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+import com.bumptech.glide.Glide;
+import com.globant.equattrocchio.cleanarchitecture.R;
+import com.globant.equattrocchio.domain.enities.ImageEntity;
+
+import java.util.List;
+
+public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.RecyclerViewViewHolder> {
+
+    private Context context;
+    private List<ImageEntity> images;
```
Comment (nacho-globant) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: change name by Image

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+    @NonNull
+    @Override
+    public RecyclerViewViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.card,parent,false);
+        return new RecyclerViewViewHolder(itemView);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerViewViewHolder holder, int position) {
+        holder.getCardImageId().setText("Image ID: " + images.get(position).getId());
```
Comment (nacho-globant) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: Replace by StringBuilder and not hardcode

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    public static class RecyclerViewViewHolder extends RecyclerView.ViewHolder{
+
+        private final TextView cardImageId;
+        private final ImageView cardImage;
+
+        public RecyclerViewViewHolder(View parent) {
+            super(parent);
+            cardImageId = itemView.findViewById(R.id.cardImageId);
```
Comment (nacho-globant) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: use ButterKnife

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+
+import com.bumptech.glide.Glide;
+import com.globant.equattrocchio.cleanarchitecture.R;
+import com.globant.equattrocchio.domain.enities.ImageEntity;
+
+import java.util.List;
+
+public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.RecyclerViewViewHolder> {
+
+    private Context context;
```
Comment (nacho-globant) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: Not keep the context

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java (hunk 14 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: ok done

File: presentation/src/main/res/layout/card.xml [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+            <TextView
+                android:id="@+id/cardImageId"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_below="@id/cardImage"
+                android:paddingLeft="5dp"
+                android:paddingRight="5dp"
+                android:paddingTop="5dp"
+                android:textColor="@color/cardview_dark_background"
+                android:text="Text"
```
Comment (nacho-globant) [inline presentation/src/main/res/layout/card.xml]: move to string

File: presentation/src/main/res/layout/card.xml [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+                android:background="?attr/selectableItemBackgroundBorderless"
+                android:clickable="true"
+                android:scaleType="centerCrop" />
+
+            <TextView
+                android:id="@+id/cardImageId"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_below="@id/cardImage"
+                android:paddingLeft="5dp"
```
Comment (nacho-globant) [inline presentation/src/main/res/layout/card.xml]: move to dimens and the rest

File: presentation/src/main/res/layout/card.xml [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+                android:paddingRight="5dp"
+                android:paddingTop="5dp"
+                android:textColor="@color/cardview_dark_background"
+                android:text="Text"
+                android:textSize="15sp"
+                android:textStyle="normal"
+                android:typeface="sans" />
```
Comment (nacho-globant) [inline presentation/src/main/res/layout/card.xml]: create a style and use it

**Review by nacho-globant — COMMENTED** [review]
some comments

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java (hunk 21 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java (hunk 18 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java (hunk 19 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java (hunk 13 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: I spoke with Nacho and finally I use Butterknife

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java (hunk 15 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/base/MainActivity.java (hunk 16 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/base/MainActivity.java]: done

File: presentation/src/main/res/layout/activity_main.xml (hunk 9 above)
Comment (SartiMau) [reply presentation/src/main/res/layout/activity_main.xml]: done

File: presentation/src/main/res/layout/card.xml (hunk 10 above)
Comment (SartiMau) [reply presentation/src/main/res/layout/card.xml]: done

Comment (SartiMau) [reply presentation/src/main/res/layout/card.xml]: done

File: presentation/src/main/res/layout/card.xml (hunk 23 above)
Comment (SartiMau) [reply presentation/src/main/res/layout/card.xml]: done

File: presentation/src/main/res/layout/card.xml (hunk 22 above)
Comment (SartiMau) [reply presentation/src/main/res/layout/card.xml]: done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java (hunk 20 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: ok

File: presentation/src/main/res/layout/card.xml (hunk 24 above)
Comment (SartiMau) [reply presentation/src/main/res/layout/card.xml]: yes, done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java (hunk 12 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/RecyclerViewAdapter.java]: done

File: data/src/main/java/com/globant/equattrocchio/data/response/Result.java [hunk 25]
```diff
@@ -8,14 +8,14 @@
 public class Result {
     @SerializedName("images")
     @Expose
-    private List<Image> mImages = null;
+    private List<ImageResponse> mImageResponses = null;
```
Comment (nacho-globant) [inline data/src/main/java/com/globant/equattrocchio/data/response/Result.java]: You should not use this code style, and not necessary asign to null (is default value) . Use  

private List<ImageResponse> imageResponses; 
or
private List<ImageResponse> images;

File: data/src/main/java/com/globant/equattrocchio/data/response/Result.java [hunk 26]
```diff
@@ -8,14 +8,14 @@
 public class Result {
```
Comment (nacho-globant) [inline data/src/main/java/com/globant/equattrocchio/data/response/Result.java]: Rename by more accord name. For example

GetImagesResponse

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/ImagesView.java (hunk 11 above)
Comment (nacho-globant) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/ImagesView.java:40]: You should not need specified a label as enabled or not. I think...

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java [hunk 27]
```diff
@@ -0,0 +1,62 @@
...
+
+import java.util.List;
+
+import butterknife.BindView;
+
+public class ImagesAdapter extends RecyclerView.Adapter<ImagesAdapter.ImagesViewHolder> {
+
+    private List<Image> images;
+
+    public ImagesAdapter(Context context, List<Image> images) {
```
Comment (nacho-globant) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java]: remove the context, if you dont need

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java [hunk 28]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    class ImagesViewHolder extends RecyclerView.ViewHolder{
+
+        @BindView(R.id.cardImageId) TextView cardImageId;
+        @BindView(R.id.cardImage) ImageView cardImage;
+
+        public ImagesViewHolder(View parent) {
+            super(parent);
+        }
```
Comment (nacho-globant) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java]: You need specified the ButterKnife.bind() here to work fine

File: presentation/src/main/res/layout/activity_main.xml [hunk 29]
```diff
@@ -14,5 +16,20 @@
             android:id="@+id/tv_incoming_json"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:text="@string/incoming_json" />
+            android:text=""
```
Comment (nacho-globant) [inline presentation/src/main/res/layout/activity_main.xml]: Not hardcode values, create a empty value in strings.xml

File: presentation/src/main/res/layout/card.xml [hunk 30]
```diff
@@ -0,0 +1,46 @@
...
+            <TextView
+                android:id="@+id/cardImageId"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_below="@id/cardImage"
+                android:paddingLeft="@dimen/card_text_padding_left"
+                android:paddingRight="@dimen/card_text_padding_right"
+                android:paddingTop="@dimen/card_text_padding_top"
+                android:textColor="@color/cardview_dark_background"
+                android:text=""
```
Comment (nacho-globant) [inline presentation/src/main/res/layout/card.xml]: the same

**Review by nacho-globant — COMMENTED** [review]
comments

File: data/src/main/java/com/globant/equattrocchio/data/response/Result.java (hunk 26 above)
Comment (SartiMau) [reply data/src/main/java/com/globant/equattrocchio/data/response/Result.java]: done

File: data/src/main/java/com/globant/equattrocchio/data/response/Result.java (hunk 25 above)
Comment (SartiMau) [reply data/src/main/java/com/globant/equattrocchio/data/response/Result.java]: ok, done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java (hunk 27 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java]: done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java [hunk 31]
```diff
@@ -0,0 +1,62 @@
...
+                    .into(holder.cardImage);
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return images.size();
+    }
+
+    class ImagesViewHolder extends RecyclerView.ViewHolder{
```
Comment (nacho-globant) [inline presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java]: You need specified as static class

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java (hunk 28 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java]: done

File: presentation/src/main/res/layout/activity_main.xml (hunk 29 above)
Comment (SartiMau) [reply presentation/src/main/res/layout/activity_main.xml]: done

File: presentation/src/main/res/layout/card.xml (hunk 30 above)
Comment (SartiMau) [reply presentation/src/main/res/layout/card.xml]: yes, done

File: presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java (hunk 31 above)
Comment (SartiMau) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java]: done

Comment (fanky10) [reply presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/view/adapters/ImagesAdapter.java]: I think there's no need for a static class here, just an inner one. It's not gonna be used outside this

File: presentation/src/main/res/layout/card.xml (hunk 30 above)
Comment (fanky10) [reply presentation/src/main/res/layout/card.xml]: I'd rather remove the android:text attribute instead, there's no need to specify it if no value is set

**Review by fanky10 — APPROVED** [review]
Looks good to me

File: presentation/src/main/res/values/styles.xml [hunk 32]
```diff
@@ -8,4 +8,10 @@
         <item name="colorAccent">@color/colorAccent</item>
     </style>
 
+    <style name="CardText" parent="TextAppearance.AppCompat.Title">
+        <item name="android:textSize">20sp</item>
```
Comment (gaston-mira-glb) [inline presentation/src/main/res/values/styles.xml]: use dimens here

**Review by gaston-mira-glb — COMMENTED** [review]
only one comment. But as Facu said, it looks good to me also.

File: presentation/src/main/res/values/styles.xml (hunk 32 above)
Comment (SartiMau) [reply presentation/src/main/res/values/styles.xml]: done

**Review by gaston-mira-glb — APPROVED** [review]

**Review by nacho-globant — APPROVED** [review]

**Pull request merged**